src/routes/vendor/vendor-show.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import fetch from 'isomorphic-fetch';
+
+import View from '../../components/vendor-show';
+
+class Vendor {
+  constructor(prev = {}, props = {}) {
+    Object.assign(this, prev, props);
+  }
+  get isRequested() { return false; }
+  get isLoading() { return false; }
+  get isLoaded() { return false; }
+  get isErrored() { return false; }
+  get isPending() { return !this.isRequested || this.isLoading; }
+
+}
+
+class PendingVendor extends Vendor {
+  get isRequested()  { return true; }
+  get isPending() { return true; }
```

Comment:
since `isPending` is computed, and essentially means "hasnot been requested, or has been requested but is still loading", setting `isLoading` here instead of strong-arming `isPending` would probably reflect the state better